File: app/src/main/java/com/example/pc/basemvp/screen/login/UserAdapter.java
package com.example.pc.basemvp.screen.login;

import android.content.Context;
import android.databinding.DataBindingUtil;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import com.example.pc.basemvp.R;
import com.example.pc.basemvp.base.recyclerView.BaseRecyclerViewAdapter;
import com.example.pc.basemvp.data.model.User;
import com.example.pc.basemvp.databinding.ItemUserBinding;

public class UserAdapter extends BaseRecyclerViewAdapter<User, UserAdapter.ItemUserViewHolder> {

    protected UserAdapter(@NonNull Context context) {
        super(context);
    }

    @NonNull
    @Override
    public ItemUserViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemUserBinding binding = DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()),
                R.layout.item_user, parent, false);
        return new UserAdapter.ItemUserViewHolder(binding, mItemClickListener);
    }

    @Override
    public void onBindViewHolder(@NonNull ItemUserViewHolder holder, int position) {
        holder.bind(getItem(position));
    }

    static class ItemUserViewHolder extends RecyclerView.ViewHolder {
        private ItemUserBinding mBinding;
        private ItemUserViewModel mViewModel;

        ItemUserViewHolder(ItemUserBinding binding, OnItemClickListener<User> itemClickListener) {
            super(binding.getRoot());
            mBinding = binding;
            mViewModel = new ItemUserViewModel(itemClickListener);
            mBinding.setViewModel(mViewModel);
        }

        void bind(User user) {
            mViewModel.setData(user);
            mBinding.executePendingBindings();
        }
    }
}

File: app/src/main/java/n/com/myapplication/screen/main/user/UserAdapter.kt
package n.com.myapplication.screen.main.user

import android.view.ViewGroup
import androidx.annotation.NonNull
import androidx.databinding.ViewDataBinding
import androidx.recyclerview.widget.RecyclerView
import n.com.myapplication.R
import n.com.myapplication.data.model.User
import n.com.myapplication.util.extension.inflateBinding
import n.com.myapplication.widget.recyclerView.BaseItemVH
import n.com.myapplication.widget.recyclerView.LoadMoreAdapter
import n.com.myapplication.widget.recyclerView.OnItemClickListener

class UserAdapter : LoadMoreAdapter<User>() {

    override fun getViewType(position: Int): Int {
        return 0
    }

    @NonNull
    override fun onCreateVH(parent: ViewGroup, viewType: Int): ItemViewHolder {
        return ItemViewHolder(listener = itemClickListener,
                binding = parent.inflateBinding(R.layout.item_user))
    }

    override fun onBindVH(holder: RecyclerView.ViewHolder, position: Int) {
        (holder as ItemViewHolder).bind(data = getItem(position))
    }

    companion object {

        class ItemViewHolder(binding: ViewDataBinding,
                listener: OnItemClickListener<User>?)
            : BaseItemVH<User>(binding, ItemUserViewModel(), listener)

    }
}